opm/autodiff/AutoDiffBlock.hpp

```diff
@@ -374,14 +378,31 @@ namespace Opm
                     jac[block] = M( D3.rows(), jac_[block].cols() );
                 }
                 else if( jac_[block].nonZeros() == 0 ) {
-                    jac[block] =  D3 * ( D1*rhs.jac_[block]) * (-1.0);
+#pragma omp critical
```

Comment:
To be sure I understand this: the #pragma applies to the following if() only, correct?